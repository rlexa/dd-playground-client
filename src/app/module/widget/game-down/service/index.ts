export * from './game-down.service';
