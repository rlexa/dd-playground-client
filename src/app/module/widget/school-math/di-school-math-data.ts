import {inject, InjectionToken} from '@angular/core';
import {jsonEqual} from 'dd-rx-state';
import {StateSubject} from 'dd-rxjs';
import {TCreatedPdf} from 'pdfmake/build/pdfmake';
import {BehaviorSubject, Observable} from 'rxjs';
import {distinctUntilChanged, filter, map, shareReplay, startWith} from 'rxjs/operators';
import {generateMathTestGrade3, MathTest} from './math-test/math-test-generator';
import {mathTestToPdf} from './math-test/math-test-pdf';

/** Current math data seed. */
export const DiSchoolMathSeed = new InjectionToken<BehaviorSubject<number>>('DI math data seed.', {
  providedIn: 'root',
  factory: () => new StateSubject<number>(1),
});

/** Current math test data. */
export const DiSchoolMathTest = new InjectionToken<Observable<MathTest>>('DI math test data.', {
  providedIn: 'root',
  factory: () =>
    inject(DiSchoolMathSeed).pipe(
      filter((seed) => seed > 0),
      map((seed) => generateMathTestGrade3({seed, title: `Mathe Arbeit #${seed}`})),
      startWith<MathTest>(null as MathTest),
      distinctUntilChanged(jsonEqual),
      shareReplay({refCount: true, bufferSize: 1}),
    ),
});

/** Current math test pdf meta data. */
export const DiSchoolMathTestPdfMeta = new InjectionToken<Observable<TCreatedPdf>>('DI math test pdf meta data.', {
  providedIn: 'root',
  factory: () =>
    inject(DiSchoolMathTest).pipe(
      map((data) => mathTestToPdf(data)),
      shareReplay({refCount: true, bufferSize: 1}),
    ),
});
