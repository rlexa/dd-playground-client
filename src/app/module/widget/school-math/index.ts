export * from './school-math.module';
