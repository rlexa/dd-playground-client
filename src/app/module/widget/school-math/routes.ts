import {Route} from '@angular/router';
import {ROUTE_ROOT, ROUTE_WILDCARD} from '../../../routing';

export default [
  {path: ROUTE_ROOT, loadComponent: () => import('./school-math.component').then((ii) => ii.SchoolMathComponent)},
  {path: ROUTE_WILDCARD, redirectTo: ROUTE_ROOT},
] as Route[];
