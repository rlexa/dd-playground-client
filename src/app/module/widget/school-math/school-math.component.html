<div ddFlexbox="hyscsw">
  <mat-card class="childhv">
    <mat-card-title>Math Test</mat-card-title>
    <mat-card-content>
      <div ddFlexbox="vys">
        <button mat-button>Generate</button>
      </div>
    </mat-card-content>
  </mat-card>

  <mat-card class="childhv">
    <mat-card-title>Output</mat-card-title>
    <mat-card-content>
      <app-simple-view [data]="data$ | async" [isWidthConstrained]="false" [isDense]="true"></app-simple-view>
    </mat-card-content>
  </mat-card>
</div>
