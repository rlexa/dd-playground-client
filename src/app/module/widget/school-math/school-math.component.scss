:host {
  display: flex;
  flex-flow: row wrap;
  place-content: flex-start;
  align-items: flex-start;

  > :nth-child(1n) {
    margin: var(--size-gap);
  }

  mat-card-content {
    display: flex;
    flex-direction: column;
  }

  p {
    display: flex;
    align-items: center;

    > :nth-child(1n + 1) {
      margin-left: var(--size-gap);
    }
  }

  label {
    font-size: var(--size-font-small);
  }

  .right {
    text-align: right;
  }
}
