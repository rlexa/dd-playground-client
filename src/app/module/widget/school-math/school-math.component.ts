import {ChangeDetectionStrategy, Component, OnDestroy, OnInit} from '@angular/core';
import {of} from 'rxjs';

@Component({
  selector: 'app-school-math',
  templateUrl: './school-math.component.html',
  changeDetection: ChangeDetectionStrategy.OnPush,
})
export class SchoolMathComponent implements OnDestroy, OnInit {
  data$ = of({title: 'Mathe Arbeit'});

  destroy() {}
  ngOnDestroy() {
    this.destroy();
  }

  ngOnInit() {}
}
