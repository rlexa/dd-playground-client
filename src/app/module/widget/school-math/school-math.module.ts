import {CommonModule} from '@angular/common';
import {NgModule} from '@angular/core';
import {MatButtonModule} from '@angular/material/button';
import {MatCardModule} from '@angular/material/card';
import {RouterModule, Routes} from '@angular/router';
import {ROUTE_ROOT, ROUTE_WILDCARD} from 'src/app/routing';
import {SimpleViewModule} from '../simple-view';
import {SchoolMathComponent} from './school-math.component';

const ROUTING: Routes = [
  {path: ROUTE_ROOT, component: SchoolMathComponent, pathMatch: 'full'},
  {path: ROUTE_WILDCARD, redirectTo: ROUTE_ROOT},
];

@NgModule({
  imports: [CommonModule, MatButtonModule, MatCardModule, SimpleViewModule, RouterModule.forChild(ROUTING)],
  exports: [SchoolMathComponent],
  declarations: [SchoolMathComponent],
})
class SchoolMathModule {}

export {SchoolMathModule, SchoolMathComponent};
