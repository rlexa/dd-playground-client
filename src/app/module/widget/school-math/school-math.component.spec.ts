import {TestBed} from '@angular/core/testing';
import {FormsModule} from '@angular/forms';
import {MatButton} from '@angular/material/button';
import {MatCard, MatCardContent, MatCardTitle} from '@angular/material/card';
import {RouterTestingModule} from '@angular/router/testing';
import {MockComponents, MockDirectives} from 'ng-mocks';
import {detectChanges, mockAll, overrideForChangeDetection} from 'src/app/test';
import {FlexboxDirective} from '../../directive/flexbox';
import {CryptoApiService} from '../../service/crypto-api';
import {SimpleViewComponent} from '../simple-view';
import {SchoolMathComponent} from './school-math.component';

describe('SchoolMathComponent', () => {
  beforeEach(() => {
    TestBed.configureTestingModule({
      imports: [RouterTestingModule, FormsModule],
      declarations: [
        SchoolMathComponent,
        MockComponents(MatButton, MatCard, MatCardTitle, MatCardContent, SimpleViewComponent),
        MockDirectives(FlexboxDirective),
      ],
      providers: [mockAll(CryptoApiService)],
    })
      .overrideComponent(SchoolMathComponent, overrideForChangeDetection)
      .compileComponents();
  });

  test('is created', () => {
    const fixture = TestBed.createComponent(SchoolMathComponent);
    detectChanges(fixture);
    expect(fixture).toMatchSnapshot();
  });
});
