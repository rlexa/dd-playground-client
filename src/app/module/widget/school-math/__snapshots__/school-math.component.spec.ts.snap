// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SchoolMathComponent is created 1`] = `
<app-school-math
  data$={[Function Observable]}
>
  <div
    ddflexbox="hyscsw"
  >
    <mat-card
      class="childhv"
    >
      <mat-card-title>
        Math Test
      </mat-card-title>
      <mat-card-content>
        <div
          ddflexbox="vys"
        >
          <button
            mat-button=""
          >
            Generate
          </button>
        </div>
      </mat-card-content>
    </mat-card>
    <mat-card
      class="childhv"
    >
      <mat-card-title>
        Output
      </mat-card-title>
      <mat-card-content>
        <app-simple-view />
      </mat-card-content>
    </mat-card>
  </div>
</app-school-math>
`;
